parser/ast/base.go

```diff
@@ -35,12 +37,22 @@ func (n *node) OriginTextPosition() int {
 }
 
 // SetText implements Node interface.
-func (n *node) SetText(text string) {
+func (n *node) SetText(enc charset.Encoding, text string) {
+	if enc == nil {
+		enc = charset.EncodingBinImpl
+	}
+	utf8Lit, _ := enc.Transform(nil, charset.HackSlice(text), charset.OpDecodeReplace)
```

Comment:
Please add a performance test result for GBK stmt.

Reply:
it definitely worse than the utf8 case, I will have the performance with the whole GBK feature

Reply:
IMO, I think this PR will reduce GBK's performance. Becaues we add one more transform for GBK. And I want to know the exact number. Just a simple test with parser only.

Reply:
like this [link](https://github.com/pingcap/tidb/pull/31039)
![image](https://user-images.githubusercontent.com/17435596/148877143-03a8781c-1fc1-40eb-b8b2-0a4be0f3ae2e.png)

Reply:
the same test case
```
BenchmarkUTF8-8   	     637	   1865371 ns/op
BenchmarkGBK-8   	     122	   9589314 ns/op
```

Reply:
Seems declined a lot. 😭

Reply:
![image](https://user-images.githubusercontent.com/20839912/148889028-38e81d70-e523-4a1a-ba37-044f8d32c6a2.png)